script.js

```diff
@@ -1,4 +1,284 @@
+var diceObject = []; // This stores the list of dice values of all players
+var combinedDiceValue = []; // This stores the combined value of the dices after an order has been determined
+var runningScore = []; // This stores the (sorted) cumulative score of each player
+var numberOfPlayers = '';
+var numberOfDice = '';
+var currentPlayer = 1;
+var nextPlayer = currentPlayer + 1;
+var currentRound = 1;
+var gameMode = 'Reset';
+var orderMode = 0; // Define 4 states (0 = Initial state, 1 = Dice 1 first, 2 = Dice 2 first, 3 = Auto highest)
+var rollMessage; // Declare this as global variable to hold the message of original dice order before sorting
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var winnerIndex;
+  var message = '';
+  var gameModeMessage =
+    'Please select a game mode (input integers 1-5) <br><br> 1. Manually select combined order; highest = winner <br> 2. Manually select combined order; lowest = winner <br> 3. Auto-generate highest combined <br> 4. Variable number of players & dice';
+  // Initialize game
+  if (gameMode == 'Reset') {
+    gameMode = 'Waiting for selection';
+    return `Welcome! ${gameModeMessage}`;
+  }
+  // Select game mode
+  if (gameMode == 'Waiting for selection') {
+    if (input >= 1 && input <= 5) {
+      gameMode = Number(input);
+      return `You have chosen mode ${gameMode}. Please click Submit to play`;
+    } else {
+      return `Input error. ${gameModeMessage}`;
+    }
+  }
+  // Game mode 1 & 2: Base; running score
+  if (gameMode == 1 || gameMode == 2) {
```

Comment:
Good to check for invalid gameMode before line 33. 
Eg. `if(gameMode < 1 || gameMode >4) { return "Input error. ${gameModeMessage}"}`